app/services/manageBusinesses/IncomeSourceRFService.scala

```diff
@@ -92,76 +76,14 @@ class IncomeSourceRFService @Inject()(val sessionService: SessionService,
-                                                                 isCrystallisedForNextTy: Boolean
-                                                               ): Option[IncomeSourceReportingFrequencySourceData] = {
-
-    val incomeSourceReportingFrequencySourceData: Option[IncomeSourceReportingFrequencySourceData] = {
-      (
-        isCrystallisedForCurrTy,
-        isCrystallisedForNextTy,
-        currentTyStatusDetailOpt.map(_.status),
-        nextTyStatusDetailOpt.map(_.status)
-      ) match {
-        case (true, false, Some(ITSAStatus.Annual), Some(ITSAStatus.NoStatus)) if currentTyStatusDetailOpt.exists(_.statusReason != StatusReason.Rollover) =>
-          Some(IncomeSourceReportingFrequencySourceData(false, true, false, false))
-        case (true, false, _, _) if isItsaStatusMandatedOrVoluntary(currentTyStatusDetailOpt) =>
-          Some(IncomeSourceReportingFrequencySourceData(false, true, false, false))
-        case (_, false, Some(ITSAStatus.Annual), _) if isItsaStatusMandatedOrVoluntary(nextTyStatusDetailOpt) =>
-          Some(IncomeSourceReportingFrequencySourceData(false, true, false, false))
-        case (false, _, _, Some(ITSAStatus.Annual)) if isItsaStatusMandatedOrVoluntary(currentTyStatusDetailOpt) =>
-          Some(IncomeSourceReportingFrequencySourceData(true, false, false, false))
-        case (false, _, _, _) if isItsaStatusMandatedOrVoluntary(currentTyStatusDetailOpt) =>
-          Some(IncomeSourceReportingFrequencySourceData(true, true, false, false))
-        case _ =>
-          None
-      }
-    }
-    incomeSourceReportingFrequencySourceData
-  }
-
-  private def redirectToRFPageChecks(latencyDetails: Option[LatencyDetails],
-                                     currentTyStatusDetailOpt: Option[StatusDetail],
-                                     nextTyStatusDetailOpt: Option[StatusDetail],
-                                     isCrystallisedForCurrTy: Boolean,
-                                     isCrystallisedForNextTy: Boolean,
-                                     incomeSourceType: IncomeSourceType,
-                                     isAgent: Boolean,
-                                     sessionData: UIJourneySessionData
-                                    )(codeBlock: UIJourneySessionData => Future[Result]): Future[Result] = {
-    val data =
-      determineIncomeSourceReportingFrequencySourceData(
-        currentTyStatusDetailOpt,
-        nextTyStatusDetailOpt,
-        isCrystallisedForCurrTy,
-        isCrystallisedForNextTy
-      )
-
-    data match {
-      case data: Option[IncomeSourceReportingFrequencySourceData] if businessIsInLatency(latencyDetails) =>
-        updateChooseTaxYearsModelInSessionData(sessionData, data)
-          .flatMap(codeBlock(_))
-      case _ =>
-        redirectToIncomeSourceAddedPage(isAgent, incomeSourceType)
-    }
+    onlyOneYearInLatency(latencyDetails) ||
+      isItsaStatusAnnual(currentTyStatus) ||
+      isItsaStatusAnnual(nextTyStatus)
```

Comment:
Looking at the `isItsaStatusAnnual` functions, if there's a missing ITSA status or the status is of No Status or something other than Annual it flags this as false.

One thought I had was that do we want the user to be able to access the latency reporting frequency pages if their ITSA status is missing or No Status? It looks like that would be the case currently

Reply:
Yeah added conditions to handle these scenarios i.e., all non quarterly status are treated as one.

For cases like ITSA status is missing and No Status is also handled. Please review.